Update normal contact force for frictional node-face penalty

recuero opened the pull request:
Closes #24632.

recuero committed:
Update normal contact force for frictional node-face penalty (#24632)

This commit changes the way the normal contact force is used in the
node-face penalty friction algorithm. The test that failed with this
change only tests a point contact for a couple of time steps. That
failure was related to a too-large penalty factor. With the previous
approach, a too-large penalty factor may not initially have a negative
influence because the algorithm was using a cached normal force value.
Slightly decreasing that value, however, makes a more extended approach
to penalty contact work for the single point test modified in this commit.

Another example of a frictional cylinder is added to avoid regressions.
That test checks for tangential slips in opposite directions. The previous
formulation is unable to run this example.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1579493/) on 978b5af wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/24643/site)

This comment will be updated on new commits.

recuero committed:
Regold node-face penalty frictional tests.

- All tests appear to exercise mostly sliding
- Former penalty factor can be higher for these cases

The second item points probably to the fact that, before,
penalty was not implementing a truly penalty formulation and the
forces were being updated with previous iteration values, some sort
of simultaneous AL. If there is interest, an implementation like that
can be implemented with its own formulation enum, i.e. not 'PENALTY'.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1580397/) on 4923ee1 wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

### Contact

<table>
  <thead>
    <tr>
      <th></th>
      <th>f43289</th>
      <th colspan=4>#24643 4923ee</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>90.10%</td>
      <td>90.10%</td>
      <td>-</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>3640</td>
      <td>3640</td>
      <td>-</td>
      <td>1</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>400</td>
      <td>400</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24643/coverage_diff/contact)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24643/coverage/contact)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/24643/coverage/xfem)

</details>



This comment will be updated on new commits.

lindsayad commented on modules/contact/test/tests/sliding_block/sliding/tests:
```diff
@@ -25,11 +25,12 @@
     max_time = 800
-#    deleted = '#6139'
+    allow_warnings = true
+    #    deleted = '#6139'
```
```suggestion
```

lindsayad commented on modules/contact/test/tests/sliding_block/sliding/tests:
```diff
@@ -38,11 +39,11 @@
     rel_err = 1e-4
     max_time = 800
-#    deleted = '#6139'
+    #    deleted = '#6139'
```
```suggestion
```

lindsayad commented on modules/contact/test/tests/verification/patch_tests/mindlin/cylinder_friction_node_face.i:
```diff
@@ -0,0 +1,339 @@
+[Kernels]
+  [TensorMechanics]
+    use_displaced_mesh = true
+    save_in = 'saved_x saved_y'
```
Do we need the save in stuff?

recuero replied:
No, we don't need it anymore.

lindsayad commented on modules/contact/test/tests/verification/patch_tests/mindlin/cylinder_friction_node_face.i:
```diff
@@ -0,0 +1,339 @@
+    order = CONSTANT
+    family = MONOMIAL
+  []
+  [saved_x]
+  []
+  [saved_y]
+  []
+  [diag_saved_x]
+  []
+  [diag_saved_y]
+  []
```
Do we need the save-in stuff?

recuero committed:
Address Alex's comments

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1584526/) on 8eb6cdf wanted to post the following:


The following coverage requirement(s) failed:
- Failed to generate `framework` coverage rate (required: 81.0%)
- Failed to generate `stochastic_tools` coverage rate (required: 88.0%)

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1584526/) on 8eb6cdf wanted to post the following:


The following coverage requirement(s) failed:
- Failed to generate `framework` coverage rate (required: 81.0%)
- Failed to generate `stochastic_tools` coverage rate (required: 88.0%)

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1584526/) on 8eb6cdf wanted to post the following:


The following coverage requirement(s) failed:
- Failed to generate `framework` coverage rate (required: 81.0%)
- Failed to generate `stochastic_tools` coverage rate (required: 88.0%)

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1584526/) on 8eb6cdf wanted to post the following:


The following coverage requirement(s) failed:
- Failed to generate `framework` coverage rate (required: 81.0%)
- Failed to generate `stochastic_tools` coverage rate (required: 88.0%)

bwspenc reviewed (approved):
Looks good.
